src/html.ml

```diff
@@ -178,7 +254,36 @@ let rec block = function
       in
       elt Block "dl" attr (Some (concat_map f l))
 
-let of_doc doc = concat_map block doc
+let of_doc ?(auto_identifiers = true) doc =
+  let identifiers = Identifiers.empty in
```

Comment:
This way of tracking looks pretty good to me! It occurs to me looking at this now that we may want to actually add these into the AST, rather than only producing them when rending HTML. But that's something we can do as a followup, by moving this into the parser.

Reply:
Do you mean adding the id attributes to the AST? If so, yes I think it would make sense.

Reply:
That's right: I mean rather than only generating IDs during output to HTML, we may want to add an option to add generated id attributes during parsing, or as a post-processing stage given a parsed AST.

This would expose the functionality both to other output formats, and to inspection by code consuming the AST.

Reply:
Sorry for the delay, I created an issue to track it https://github.com/ocaml/omd/issues/296